### PR TITLE
LSIF: Add processedAt to dumps.

### PR DESCRIPTION
**Pull request by efritz**
Our old uploaded_at field was the time the record was created, which is NOT the time it was uploaded. Rename this to processed_at and create a new column uploaded_at that will be the time that the convert job was queued.

### CODE REVIEW

File: migrations/1528395615_lsif_processed_at.up.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+BEGIN;
+
+ALTER TABLE lsif_dumps RENAME uploaded_at to processed_at;
+ALTER TABLE lsif_dumps ADD COLUMN uploaded_at timestamp with time zone;
+UPDATE lsif_dumps SET uploaded_at = processed_at;
+ALTER TABLE lsif_dumps ALTER COLUMN uploaded_at SET NOT NULL;
+DROP INDEX lsif_dumps_uploaded_at;
+CREATE INDEX lsif_dumps_uploaded_at ON lsif_dumps(uploaded_at);
```
Comment (unknwon) [inline migrations/1528395615_lsif_processed_at.up.sql]: Just wonder, do we have an INDEX for the old "uploaded_at" column (that is renamed to "processed_at" in this script) that has the same name of "lsif_dumps_uploaded_at"?

File: migrations/1528395615_lsif_processed_at.down.sql [hunk 2]
```diff
@@ -0,0 +1,7 @@
+BEGIN;
+
+ALTER TABLE lsif_dumps DROP COLUMN uploaded_at;
```
Comment (unknwon) [inline migrations/1528395615_lsif_processed_at.down.sql]: Do we need to drop the INDEX "lsif_dumps_uploaded_at", or it's dropped altogether with this line?

Comment (efritz) [reply migrations/1528395615_lsif_processed_at.down.sql]: Indices should be dropped with the column.

File: migrations/1528395615_lsif_processed_at.up.sql (hunk 1 above)
Comment (efritz) [reply migrations/1528395615_lsif_processed_at.up.sql]: So we had an index lsif_dumps_uploaded_at on the uploaded_at column. We renamed the column, so the index name remains the same but the target changes. We still want the index on the uploaded_at column as we want to delete the oldest dumps when we need to clean it up (that's the only query that uses this index).

**Pull request merged**